webpack.config.js

```diff
@@ -236,6 +233,28 @@ function getWebpackConfig( { cssFilename, externalizeWordPressPackages = false }
 						},
 					],
 				},
+				{
+					test: /\.jsx?$/,
+					include: /node_modules\//,
+					exclude: /@babel(?:\/|\\{1,2})runtime/,
```

Comment:
There are several libraries that we can trust to be ES5:
- `babel-runtime` -- the 6 version used still by many libraries: `react-virtualized`, `click-outside`, `wpcom` (!)
- `core-js`
- everything in `@wordpress/*`
- `lodash`
- `regenerate-unicode-properties` -- used by the `@babel/plugin-proposal-unicode-property-regex` transform
- `gridicons`
- `fbjs`
- `react` and related
- `wpcom`

Reply:
One strange file that gets transpiled by Babel:
```
node_modules/css-loader/lib/css-base.js
```
I don't know why exactly, but it's probably the JS code that's added by `css-loader` and then removed again by the extract plugin?

Reply:
Looks like `exclude` can be a function instead of a regex. Perhaps that would make building and maintaining that list easier. https://webpack.js.org/configuration/module/#condition

Reply:
btw, `react` is not safe. They use Map and Set. :) https://reactjs.org/docs/javascript-environment-requirements.html

Reply:
> btw, react is not safe. They use Map and Set.

That is safe for us. We import `@babel/polyfill` at the very beginning, and it polyfills the `Map` and `Set` globals, according to the browser requirements in `@babel/preset-env`.

It wouldn't be safe if we:
- used `useBuiltins: 'usage'` in `preset-env`. Babel needs to transpile a module to see that a polyfill is needed.
- used `corejs: true` in `transform-runtime`. Then Babel transpiles usages of the `Map` global to imports from `babel-runtime-corejs`, avoiding the need for global namespace pollution.

In both cases, Babel would need to see the React module to polyfill correctly.

Reply:
switched to a function based exclude in https://github.com/Automattic/wp-calypso/pull/30196/commits/40d46b81e3a71a1b87791a587d11882d6fd17803

also looked at trying to automate the set of includes with a function that parses each package.json and checks the `engines` property, but it looks like that won't fly because at least some of the packages that have dropped node 4 support have not updated `engines`.